### PR TITLE
switch to atcute's packages for record verification

### PR DESCRIPTION
**Pull request by mary-ext**
The `validateRecord` function provides an array of errors indicating what went wrong, but it's not hooked up to anything yet

### CODE REVIEW

File: src/utils/api.ts [hunk 1]
```diff
@@ -1,48 +1,79 @@
...
+  XrpcHandleResolver,
+} from "@atcute/identity-resolver";
+
 import { setPDS } from "../components/navbar";
-import { DidDocument } from "@atcute/client/utils/did";
-import { createStore } from "solid-js/store";
 
-const didPDSCache: Record<string, string> = {};
-const [labelerCache, setLabelerCache] = createStore<Record<string, string>>({});
+export const didDocumentResolver = new CompositeDidDocumentResolver({
```
Comment (notjuliet) [inline src/utils/api.ts:19]: why is this defined here and in `src/utils/resolvers.ts`? can't it be reused?

**Comment by notjuliet** [comment]
currently breaks trying to load a labeler's repo page (despite being able to fetch the did document) and it shows actually valid records as invalid (i've actually only noticed the profile record being marked as valid, all other lexicons fail)

image
image

**Comment by notjuliet** [comment]
also you can remove `vite-plugin-wasm` since its not used anymore

File: src/utils/api.ts [hunk 2]
```diff
@@ -1,48 +1,79 @@
...
+    throw new Error(`Invalid DID identifier`);
+  }
+
+  const didDoc = await didDocumentResolver.resolve(did);
+
+  return didDoc;
+}, "resolveDidDocument");
+
 const getPDS = query(async (did: string) => {
-  if (did in didPDSCache) return didPDSCache[did];
```
[the remark this answers is not in the document]
Comment (mary-ext) [reply src/utils/api.ts:11]: does `query` not cache this? my assumption is that it would work even when async so long as there's no `await` prior to being called

[the remark this answers is not in the document]
Comment (notjuliet) [reply src/utils/api.ts:11]: never managed to make it work for more than a few seconds at most, then it would refetch, eventually i just settled for David's solution

[the remark this answers is not in the document]
Comment (mary-ext) [reply src/utils/api.ts:11]: hmmm that's unfortunate. there's a proper fix but it also requires a bunch of changes to how you're currently doing layouting atm, seems more appropriate as a separate PR so I'll try doing what I can here to get it to work as before

File: src/utils/api.ts (hunk 1 above)
Comment (mary-ext) [reply src/utils/api.ts:19]: `resolvers.ts` was a leftover file, going to delete that one instead

**Review by notjuliet — COMMENTED** [review]
seems fine and without the bugs i encountered before
worth noting there are differences with the previous public-transport implementation but i'm not sure where the protocol stand on those edge cases, possibly this will need fixes?

examples:
https://pdsls.dev/at/did:plc:t423oqsrtl5gtgdfxvye5gco/com.example.test/3lfrr6gibx4lk (invalid with public-transport, valid with atcute)
https://pdsls.dev/at/did:plc:t423oqsrtl5gtgdfxvye5gco/app.bsky.feed.post/3lfptmt2eqwh2 (valid with public-transport, invalid with atcute)

**Review by notjuliet — APPROVED** [review]

**Comment by mary-ext** [comment]
Seems to be byte serialization related, the first one is correct but I'll check the second one later
